File: apps/server/src/auth/dtos.rs
use crate::auth::UserView;
use jiff::Timestamp;
use serde::{Deserialize, Serialize};
use validator::Validate;

#[derive(Debug, Deserialize, Validate)]
pub struct LoginDto {
	#[validate(length(
		min = 1,
		max = 255,
		message = "El correo electrónico es obligatorio y debe tener entre 1 y 255 caracteres."
	))]
	pub email: String,

	#[validate(length(
		min = 1,
		max = 255,
		message = "La contraseña es obligatoria y debe tener entre 1 y 255 caracteres."
	))]
	pub password: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LoginResponse {
	pub user: UserView,
	pub access_token: String,
	pub access_token_exp: Timestamp,
	pub refresh_token_exp: Timestamp,
	pub refresh_token: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RefreshResponse {
	pub access_token: String,
	pub access_token_exp: Timestamp,
}

File: apps/client/src/lib/auth/dtos.ts
import * as v from "valibot"

export const loginDTOSchema = v.object({
	email: v.pipe(
		v.string(),
		v.trim(),
		v.minLength(1, "El email es obligatorio."),
		v.maxLength(255, "El email no puede tener más de 255 caracteres."),
	),
	password: v.pipe(
		v.string(),
		v.minLength(1, "La contraseña es obligatoria."),
		v.maxLength(255, "La contraseña no puede tener más de 255 caracteres."),
	),
})

export type LoginDTOSchema = typeof loginDTOSchema
export type LoginDTO = v.InferInput<typeof loginDTOSchema>
